src/components/MovieListPage/MovieListPage.tsx

```diff
@@ -0,0 +1,235 @@
+  const [total, setTotal] = useState<number>();
+
+  const fetchData = async (params?: MovieQueryParams) => {
+    try {
+      if (cancelSource) {
+        cancelSource.cancel();
+        setCancelSource(null);
+      }
+      const source = axios.CancelToken.source();
+      setCancelSource(source);
+      const result = await new MoviesService().get(params, source);
+      setCancelSource(null);
+      setMovieList(result.data);
+      setTotal(result.totalAmount);
+    } catch (error) {
+      if (axios.isCancel(error)) return;
+    }
+  };
+
+  const [dialogContent, setDialogContent] = useState<DialogProps>({
+    children: null,
+    title: '',
+  });
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  useEffect(() => {
+    if (isMount) return;
+    fetchData({
+      search: searchTerm,
+      searchBy: 'title',
+      sortBy: sort,
+      sortOrder: 'asc',
+    });
+  }, [searchTerm]);
+
+  useEffect(() => {
+    if (isMount) return;
+    fetchData({
+      search: genre,
+      searchBy: 'genres',
+      sortBy: sort,
+      sortOrder: 'asc',
+    });
+  }, [genre]);
+
+  useEffect(() => {
+    if (isMount) return;
+    fetchData({
+      search: searchTerm || genre,
+      searchBy: searchTerm ? 'title' : 'genres',
+      sortBy: sort,
+      sortOrder: 'asc',
+    });
+  }, [sort]);
+
+  function toggleDialog(isOpen: boolean) {
+    setShowDialog(isOpen);
+  }
+
+  function handleMovieEdit(movie: Movie) {
+    setDialogContent({
+      title: 'EDIT MOVIE',
+      children: (
+        <MovieForm
+          movie={movie}
+          onSubmit={(mov) => {
+            setShowDialog(false);
+          }}
+        ></MovieForm>
+      ),
+    });
+    setShowDialog(true);
+  }
+
+  function handleMovieDeleteClick(movie: Movie) {
+    setDialogContent({
+      title: 'DELETE MOVIE',
+      children: (
+        <DialogContent>
+          <span>Are you sure you want to delete this movie?</span>
+          <ConfirmButton onClick={() => handleConfirmMovieDelete(movie)}>
+            CONFIRM
+          </ConfirmButton>
+        </DialogContent>
+      ),
+    });
+    setShowDialog(true);
+  }
+
+  function handleConfirmMovieDelete(movie: Movie) {
+    toggleDialog(false);
+  }
+
+  function handleAddMovie() {
+    setDialogContent({
+      title: 'EDIT MOVIE',
+      children: (
+        <MovieForm
+          onSubmit={(mov) => {
+            setShowDialog(false);
+          }}
+        ></MovieForm>
+      ),
+    });
+    setShowDialog(true);
+  }
+
+  function handleSetGenre(genre: string) {
+    if (genre === 'All') return setGenre('');
+    setGenre(genre);
+  }
+  return (
+    <MovieListPageContainer>
+      <TopContainer>
+        {selectedMovie ? (
+          <DetailsContainer>
+            <TopContainerHeader>
+              <AppLogo />
+              <SearchSwitcherButton onClick={() => setSelectedMovie(undefined)}>
+                <Icon src={'/images/svg/magnifying-glass-svgrepo-com.svg'} />
```

Comment:
Please do not set 'magic strings'.
Instead import it as image and use import.
Like this:
`import SearchLogoIcon from '/images/svg/magnifying-glass-svgrepo-com.svg'`

`<Icon src={SearchLogoIcon} />`